src/compiler/parser.ts

```diff
@@ -3337,7 +3337,10 @@ namespace ts {
         function parseAwaitExpression() {
             const node = <AwaitExpression>createNode(SyntaxKind.AwaitExpression);
             nextToken();
-            node.expression = parseSimpleUnaryExpression();
+            node.expression = token === SyntaxKind.AwaitKeyword
+                // Forbid `await await`

```

Comment:
Why is `await await` forbidden?

* *UnaryExpression*
* *AwaitExpression*
* `await` *UnaryExpression*
* `await` *AwaitExpression*
* `await` `await` *UnaryExpression*